### PR TITLE
[ENH] M4 Forecasting Competition Catalogue

### LINKED ISSUE #9077: [ENH] Catalogues for Different Benchmarking Studies and Competitions
PR #8749 adds catalogue API, it includes the `BaseCatalogue` class and a dummy forecasting and classification catalogue along with tests. Details about the API can be found on the above mentioned PR.

We should add popular benchmarking studies and competition as concrete catalogue classes. We can start with this non-exhaustive list:

**Forecasting**:

- [ ] M4 Competition
- [ ] M5 Competition

**Classification**:

- [ ] TSC Bake off
- [ ] Bake off redux

I don't think implementation is the challenging part here, what's challenging is reading the papers or reports from the competitions and studies and finding equivalent algorithms present in sktime. If there is no equivalent algorithm present in sktime, then an issue should be opened for it.

### PR DESCRIPTION
**Pull request by jgyasu**
This PR adds the M4 Competition catalogue to sktime.

M4 consists of datasets across six frequencies (Yearly, Quarterly, Monthly, Weekly, Daily, Hourly). Since the estimators and metrics used in M4 require the seasonal periodicity (sp) to be specified, this PR introduces a base M4 catalogue class and six child classes, one for each frequency, where sp is fixed appropriately.

The original M4 competition included:

* 9 statistical benchmark methods
* 2 simple machine learning benchmarks: (1) a Multilayer Perceptron (MLP) and (2) a Recurrent Neural Network (RNN)

This catalogue includes the statistical benchmarks only. The two ML benchmarks are not included because the specific architectures used in M4 are not currently implemented as sktime estimators. Adding those (and potentially the top three winning methods from M4) would be valuable, but is out of scope for this PR.

M4 used a fixed-origin holdout evaluation strategy (single train/test split per series) with frequency-dependent forecast horizons, so I have intentionally not added any splitting strategy to the catalogue. The benchmark metric used in the competition was sMAPE, MASE, and OWA (Overall Weighted Average), which is introduced to sktime in PR #9111.

Towards #9077
Stacks on #9111

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Failures are related to the catalogue - could you check what is happening there?

**Comment by jgyasu** [comment]
All the failing tests are due to sporadic data download failures except `run-notebook-examples` which fails because notebook can't be run in the CI, it contains a very large experiment.

**Comment by fkiraly** [comment]
can you change the notebook so it does not run a large experiment? E.g., by commenting out parts, running a smaller experiment, etc.

**Review by fkiraly — CHANGES_REQUESTED** [review]
There are a lot of conflicts - could you kindly go through the PR and see what is easily resolvable, and whether there are any issues that have arisen due to wrong merge order?

**Review by fkiraly — CHANGES_REQUESTED** [review]
The tutorial fails to run without exceptions, can you plase check?

**Review by fkiraly — APPROVED** [review]
I think this is great!
